Sources/Plasma/Apps/plClient/Mac-Cocoa/PLSLoginController.mm

```diff
@@ -0,0 +1,183 @@
+/*==LICENSE==*
+
+CyanWorlds.com Engine - MMOG client, server and tools
+Copyright (C) 2011  Cyan Worlds, Inc.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#include <regex>
+
+#include "PLSLoginController.h"
+
+#include "hsEndian.h"
+#include "NSString+StringTheory.h"
+#include "pfPasswordStore/pfPasswordStore.h"
+#include "PLSServerStatus.h"
+#include "plNetGameLib/plNetGameLib.h"
+#include "plProduct.h"
+#include "plNetClient/plNetClientMgr.h"
+#include "pnEncryption/plChallengeHash.h"
+
+#define FAKE_PASS_STRING @"********"
+
+static NSOperationQueue* _loginQueue = nil;
+
+@implementation PLSLoginController
+
++ (void)initialize
+{
+    _loginQueue = [NSOperationQueue new];
+    _loginQueue.maxConcurrentOperationCount = 1;
```

Comment:
I stared at this for a bit - an operation queue is overkill but _is proper_ in this situation. NetComm is a singleton, so we need a singleton operation queue to safely wrap access. The only part that's really overkill is I think it would be difficult to have several requests going at once that would require an operation queue to control access to the singleton. But it could be helpful when debugging SwiftUI.